pootle/static/js/shared/components/Modal.js

```diff
@@ -58,8 +82,14 @@ const Modal = React.createClass({
   /* Handlers */
 
   handleClose() {
+    if ('onCanClose' in this.props && !this.props.onCanClose()) {
```

Comment:
This doesn't have much sense in a declarative UI IMHO.

There is some UI state which allows/disallows performing actions at any point in time. _Closing_ (or _opening_) a modal is no more than a reflection of state/props and determining whether it should be displayed or not. There is already the `onClose` handler which allows for such interaction (`UserProfileEdit` has an example), so this `onCanClose` besides confusing, is not needed.